File: gaia-sdk-python/gaia_sdk/graphql/request/input/UpdateSkillImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class UpdateSkillImpulse():
    """
    The specification to update a Skill instance
    """
    tenantId: str
    reference: str
    qualifier: str
    appendent: str
    labelList: List[str]
    repositoryUri: str

    def __init__(self, tenantId: str, reference: str, qualifier: str, appendent: str, labelList: List[str], repositoryUri: str):
        self.tenantId = tenantId
        self.reference = reference
        self.qualifier = qualifier
        self.appendent = appendent
        self.labelList = labelList
        self.repositoryUri = repositoryUri

    def __eq__(self, other):
        if type(other) is type(self):
            return self.tenantId == other.tenantId and self.reference == other.reference and self.qualifier == other.qualifier and self.appendent == other.appendent and self.labelList == other.labelList and self.repositoryUri == other.repositoryUri
        return False

    def __repr__(self):
        return {'tenantId': self.tenantId, 'reference': self.reference, 'qualifier': self.qualifier, 'appendent': self.appendent, 'labelList': self.labelList, 'repositoryUri': self.repositoryUri}

File: gaia-sdk-javascript/src/graphql/request/input/UpdateSkillImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to update a Skill instance
 */
export class UpdateSkillImpulse {
public _typeName = "UpdateSkillImpulse";
    private tenantId:Uuid;
    private reference:Uuid;
    private qualifier:String;
    private appendent:String;
    private labelList:Array<String>;
    private repositoryUri:String;
    private repositoryType:String;

    constructor (tenantId:Uuid, reference:Uuid, qualifier:String, appendent:String, labelList:Array<String>, repositoryUri:String, repositoryType:String) {
        this.tenantId = tenantId;
        this.reference = reference;
        this.qualifier = qualifier;
        this.appendent = appendent;
        this.labelList = labelList;
        this.repositoryUri = repositoryUri;
        this.repositoryType = repositoryType;
    }
}
